src/services/Frontline.js

```diff
@@ -124,6 +135,111 @@ class Frontline {
       }
     }
   }
+
+  /**
+   * Load Infinity
+   */
+
+  httpRequestWrapper = (method, URL) => {
+    return new Promise((resolve, reject) => {
+      const xhr_obj = new XMLHttpRequest();
+      xhr_obj.responseType = "json";
+      xhr_obj.open(method, URL);
+      xhr_obj.onload = () => {
+        const data = xhr_obj.response;
+        resolve(data);
+      };
+      xhr_obj.onerror = () => {
+        reject("failed");
+      };
+      xhr_obj.send();
+    });
+  };
+
+  async getProducts(apiEndpoint, params) {
+    return await this.httpRequestWrapper(
+      "GET",
+      `${apiEndpoint}/products.json?${params}`
+    );
+  }
+
+  footerHeight = () => {
+    var sum = 0;
+    LOAD_INFINITY_CLASSES.footer.split(",").forEach((className) => {
+      var element = document.querySelector(className);
+      if (typeof element != "undefined" && element != null) {
+        sum += element.offsetHeight;
+      }
+    });
+    return sum;
+  };
+
+  populateUI = (newProducts) => {
+    var products = document.querySelector(LOAD_INFINITY_CLASSES.appendProducts);
+
+    products.insertAdjacentHTML("beforeend", newProducts);
+    document.querySelector(LOAD_INFINITY_CLASSES.loadMore).style.display =
+      "none";
+  };
+
+  pagination() {
+    document.addEventListener("DOMContentLoaded", () => {
+      var page = 1;
+      var loadingNow = false;
+
+      var categoryId = document
+        .querySelector(LOAD_INFINITY_CLASSES.category)
+        .getAttribute("data-id");
+
+      if (categoryId) {
+        window.addEventListener("scroll", () => {
+          var categoryPage = document.querySelector(
+            LOAD_INFINITY_CLASSES.category
+          );
+          var loadMore = document.querySelector(LOAD_INFINITY_CLASSES.loadMore);
+
+          if (
+            typeof categoryPage != "undefined" &&
+            categoryPage != null &&
+            typeof loadMore != "undefined" &&
+            loadMore != null
+          ) {
+            const { designId, storeId, apiEndpoint } = this.options;
+
+            if (
+              window.innerHeight + window.scrollY >=
+              document.body.offsetHeight - this.footerHeight()
+            ) {
+              if (loadingNow) {
+                return;
+              }
+              document.querySelector(
+                LOAD_INFINITY_CLASSES.loadMore
+              ).style.display = "block";
+              loadingNow = true;
+
+              var params = `store_id=${storeId}&category_id=${categoryId}&page=${
```

Comment:
You can also use `URLSearchParams` to safely construct query strings with proper encoding etc.

```ts
const params = new URLSearchParams()
params.set('store_id', storeId)
...
params.toString()
```